File: PJH/완전탐색/prog_level1_최소직사각형.swift
//
//  prog_level1_최소직사각형.swift
//  PJH
//
//  Created by ParkJunHyuk on 2023/11/19.
//

import Foundation

func solution(_ sizes:[[Int]]) -> Int {
    var size = sizes
    let count = size.count;
    var x: [Int] = []
    var y: [Int] = []

    for i in 0..<count {
        size[i].sort()
        x.append(size[i][0])
        y.append(size[i][1])
    }
    return x.max()! * y.max()!
}

File: YSJ/완전탐색/prog_level1_최소직사각형.js
function solution(sizes) {
    let sorted = [...sizes]
    
    sorted.forEach(v => {
        v.sort((a,b) => a-b)
    })
    
    
    let maxSize = [0,0];
    sorted.forEach(([w,h]) => {
        if (w > maxSize[0]) maxSize[0] = w;
        if (h > maxSize[1]) maxSize[1] = h;
    }) 
    
    
    return maxSize[0] * maxSize[1];
}